DataFormats/MuonReco/interface/MuonEnergy.h

```diff
@@ -55,32 +64,37 @@ namespace reco {
        /// RecHits
        ///
        /// crossed HCAL rechits
-       const std::vector<HBHERecHit>* getHadRecHits() const
+       std::vector<HcalMuonRecHit> crossedHadRecHits;
+       bool crossedHadRecHitsValid;
+
+       const std::vector<HcalMuonRecHit>* getCrossedHadRecHits() const

```

Comment:
this method is not very useful, considering that the data is public.
Other cases in Muon.h with xyzValid flags, imply that data is accessible directly and a user should check the flag separately (e.g. ```reco::Muon::matches```).
I suggest to keep the same logic and keep the struct as simple as possible.

Reply:
ok